framework/include/reporters/ReporterContext.h

```diff
@@ -417,9 +391,12 @@ class ReporterGeneralContext : public ReporterContext<T>
    */
   virtual void declareVectorClone(ReporterData & r_data,
                                   const ReporterName & r_name,
-                                  const ReporterMode & mode) const override;
+                                  const ReporterMode & mode,
+                                  const MooseObject & producer) const override;
 
   virtual void resize(dof_id_type local_size) final;
+
+  virtual std::string contextType() const override { return MooseUtils::prettyCppType(this); }
```

Comment:
Why can't this just be in ReporterContext?

Reply:
Because `this` changes depending on what context you have. If you put this just in `ReporterContext`, `this` will always be of type `ReporterContext<T>`, even if you call `ReporterGeneralContext<T>.contextType()`.

Reply:
Ah yes, stupid C++. What about making this an input to the Constructor instead? It just seems silly to have the same method in each. I am thinking something like.

```
Context(..., const std::string & context_type = ""

NewContext : Context(..., MooseUtils::prettyCppType<NewContext>());

```

Reply:
This was how I had it originally. I'm fine with that. Will move

Reply:
Actually... this ends up adding quite a bit more code compared to just having the virtuals. I'm at a loss